packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/AccessorModifierIrGeneration.kt

```diff
@@ -616,16 +649,29 @@ class AccessorModifierIrGeneration(private val pluginContext: IrPluginContext) {
     }
 
     @Suppress("ReturnCount")
-    private fun getListGenericCoreType(declaration: IrProperty): CoreType? {
+    private fun getCollectionGenericCoreType(
+        collectionType: CollectionType,
+        declaration: IrProperty
+    ): CoreType? {
         // Check first if the generic is a subclass of RealmObject
         val descriptorType = declaration.symbol.descriptor.type
-        val listGenericType = descriptorType.arguments[0].type
-        if (inheritsFromRealmObject(listGenericType.constructor.supertypes)) {
+        val collectionGenericType = descriptorType.arguments[0].type
+
+        // No embedded objects for sets
+        val supertypes = collectionGenericType.constructor.supertypes
+        if (collectionType == CollectionType.SET && inheritsFromEmbeddedRealmObject(supertypes)) {
+            logError(
+                "Error in field ${declaration.name} - ${collectionType.name}s do not support embedded realm objects element types.",
```

Comment:
I'm not sure we should use `collectionType.name` directly here as it will result in some weird looking and internal looking names. This will turn into "SETs do not support`.... I would add a `description` or something to this enum and then use that and then set that to the public API name, so e.g `collectionType.description` would turn into `RealmSet`

Reply:
Same in other error messages

Reply:
Yes, good point 👍